instrumentor/sources_webhooks.go

```diff
@@ -0,0 +1,263 @@
+			new.Labels[consts.WorkloadKindLabel],
+			"Source workload-kind label is immutable",
+		))
+	}
+	if new.Labels[consts.WorkloadNameLabel] != old.Labels[consts.WorkloadNameLabel] {
+		allErrs = append(allErrs, field.Invalid(
+			field.NewPath("metadata").Child("labels"),
+			new.Labels[consts.WorkloadNameLabel],
+			"Source workload-name label is immutable",
+		))
+	}
+	if new.Labels[consts.WorkloadNamespaceLabel] != old.Labels[consts.WorkloadNamespaceLabel] {
+		allErrs = append(allErrs, field.Invalid(
+			field.NewPath("metadata").Child("labels"),
+			new.Labels[consts.WorkloadNamespaceLabel],
+			"Source workload-namespace label is immutable",
+		))
+	}
+	if new.Spec.Workload != old.Spec.Workload {
+		allErrs = append(allErrs, field.Invalid(
+			field.NewPath("spec").Child("workload"),
+			new.Spec.Workload,
+			"Source workload is immutable",
+		))
+	}
+
+	errs := s.validateSourceFields(ctx, new)
+	if len(errs) > 0 {
+		allErrs = append(allErrs, errs...)
+	}
+
+	if len(allErrs) == 0 {
+		return nil, nil
+	}
+
+	return nil, apierrors.NewInvalid(
+		schema.GroupKind{Group: "odigos.io", Kind: "Source"},
+		new.Name, allErrs)
+}
+
+func (s *SourcesValidator) ValidateDelete(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+	return nil, nil
+}
+
+func (s *SourcesValidator) validateSourceFields(ctx context.Context, source *v1alpha1.Source) field.ErrorList {
+	allErrs := make([]*field.Error, 0)
+
+	if source.Labels[consts.WorkloadNameLabel] != source.Spec.Workload.Name {
+		allErrs = append(allErrs, field.Invalid(
+			field.NewPath("metadata").Child("labels"),
+			source.Labels[consts.WorkloadNameLabel],
+			"odigos.io/workload-name must match spec.workload.name",
+		))
+	}
+
+	if source.Labels[consts.WorkloadNamespaceLabel] != source.Spec.Workload.Namespace {
+		allErrs = append(allErrs, field.Invalid(
+			field.NewPath("metadata").Child("labels"),
+			source.Labels[consts.WorkloadNamespaceLabel],
+			"odigos.io/workload-namespace must match spec.workload.namespace",
+		))
+	}
+
+	if source.Labels[consts.WorkloadKindLabel] != string(source.Spec.Workload.Kind) {
+		allErrs = append(allErrs, field.Invalid(
+			field.NewPath("metadata").Child("labels"),
+			source.Labels[consts.WorkloadKindLabel],
+			"odigos.io/workload-kind must match spec.workload.kind",
+		))
+	}
+
+	err := s.validateSourceUniqueness(ctx, source)
+	if err != nil {
+		allErrs = append(allErrs, field.Invalid(
+			field.NewPath("spec").Child("workload"),
+			source.Spec.Workload,
+			err.Error(),
+		))
+	}
+
+	if source.Spec.Workload.Kind == "Namespace" &&
+		(source.Spec.Workload.Name != source.Spec.Workload.Namespace) {
+		allErrs = append(allErrs, field.Invalid(
+			field.NewPath("spec").Child("workload").Child("namespace"),
+			source.Spec.Workload.Namespace,
+			"namespace Source must have matching workload.name and workload.namespace",
+		))
+	}
+
+	workloadKind := workload.WorkloadKindFromString(string(source.Spec.Workload.Kind))
```

Comment:
This will allow the kind to be with any casing (e.g. `Deployment`, `deployment`, `DEPLOYMENT`). do you think we should enforce a specific casing so that we don't end up with mismatches downstream when we try to use it?

Reply:
Good point. Updated the validating webhook to enforce consistent casing on admission